### PR TITLE
Improve deployment

### PR DESCRIPTION
**Pull request by henry410213028**
## Types of Changes

- **Bugfix**
- **Refactoring**
- **Documentation Update**

## Description

The main purpose of this PR is to improve our Airflow setup process, fix some errors, and make it easier for new contributors to use.

Below is the list of changes:

- Airflow Configuration
  - Change `AIRFLOW_HOME` from "/usr/local/airflow" to "/opt/airflow" because we have switched to the official Airflow Python 3.8 Docker image.
  - Disable the example DAGs.

- DAG files
  - Fix `AIRFLOW_HOME` related file path issues in some DAG tasks.

- Python dependencies
  - Use `pip` to manage dependencies, not use `poetry`.
  - Update `requirements.txt` and add Airflow constraints file.
  - Remove `poetry` configuration.

- Update Dockerfile
  - Install dependencies via `pip` instead of `poetry`.
  - Use `entrypoint.sh` to allow adding initialization tasks before starting Airflow services.
  - Remove unused `Dockerfile.test`.

- Use docker-compose to deploy Airflow services
  - Update `Makefile` and add related aliases to make it easier to use.
  - Use an external Docker volume to store Airflow data to prevent data loss when restarting the container.

- Update documentation
  - Remove outdated information in `README.md`.
  - Move some information to `CONTRIBUTING.md`, `MAINTENANCE.md`, and `DEPLOYMENT.md` to make it easier to understand how to contribute, maintain, and deploy.

- Remove unused Node.js configuration files.

## Steps to Test This Pull Request

Follow the steps for setting up the dev/test environment in `README.md`.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+# Dev dependencies
+bandit
+black
+flake8
+isort
+mypy
+pytest
+pytest-cov
+safety
```
Comment (Lee-W) [inline requirements.txt:22]: Should we separate it as `requirement-dev.txt` for local development testing?

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+# Editable install with no version specified.
+google-cloud-bigquery
+pandas
+pyarrow
+pydantic<2
+pygsheets
+requests
+searchconsole
+StrEnum
```
Comment (Lee-W) [inline requirements.txt:12]: Is `StrEnum` required? Could we use the trick here? https://docs.python.org/3.8/library/enum.html#others

File: pyproject.toml [hunk 3]
```diff
@@ -20,6 +20,15 @@ pygsheets = "^2.0.6"
...
+azure-mgmt-datalake-store = ">=0.5.0"
+# https://stackoverflow.com/questions/68687548/apache-airflow-airflow-initdb-throws-modulenotfounderror-no-module-named-wtf
+marshmallow = "2.21.0"
+wtforms = "2.3.3"
+# https://stackoverflow.com/questions/66774109/install-airflow-importerror-no-module-named-clsregistry
+sqlalchemy = "1.3.23"
+flask-sqlalchemy = "2.4.4"
 
 [tool.poetry.dev-dependencies]
 safety = "^1.9.0"
```
Comment (Lee-W) [inline pyproject.toml:34]: I thought we were removing `poetry`. am I miss anything?

File: constraints-3.8.txt [hunk 4]
```diff
@@ -0,0 +1,349 @@
+# for apache-airflow==1.10.13
```
Comment (Lee-W) [inline constraints-3.8.txt:1]: I would like to confirm where it comes from. Airflow Official

File: .env.template [hunk 5]
```diff
@@ -1,4 +1,4 @@
-# get the complete env from other volunteers, please
+AIRFLOW_HOME=/opt/airflow
 BIGQUERY_PROJECT=pycontw-225217
```
Comment (Lee-W) [inline .env.template:2]: nitpick: I feel this might be something we should make it configurable.  🤔

File: README.md [hunk 6]
```diff
@@ -1,176 +1,89 @@
...
+- [Contact](#contact)
 
-## Year to Year Jobs
+## Prerequisites
 
-這三個 job 什麼時候打開需要人工確認（麻煩當年的組長大大了），理論上是售票開始前我們要測試一下然後打開
+- [Python 3.8+](https://www.python.org/downloads/release/python-3811/)
+- [Docker](https://docs.docker.com/get-docker/)
+- [Git](https://git-scm.com/book/zh-tw/v2/%E9%96%8B%E5%A7%8B-Git-%E5%AE%89%E8%A3%9D%E6%95%99%E5%AD%B8)
+- [Poetry](https://python-poetry.org/docs/#installation) (Optional, only for creating virtual environments during development)
```
Comment (Lee-W) [inline README.md:24]: Should we add instruction on how to `virtualenv` instead? or use tools like `pip-tools`, `Pipenv`. AFAIK, `poetry` does not use `pip` to resolve dep. I would suggest we either use `poetry` for all env or not use it at all

File: dags/ods/survey_cake/dags/questionnaire_2_bigquery.py [hunk 7]
```diff
@@ -27,7 +29,8 @@
 )
 with dag:
     if bool(os.getenv("AIRFLOW_TEST_MODE")):
-        FILENAMES: Dict[str, Dict] = {"fixtures/data_questionnaire.csv": {}}
+        filepath = Path(AIRFLOW_HOME) / "dags/fixtures/data_questionnaire.csv"
+        FILENAMES: Dict[str, Dict] = {str(filepath): {}}
```
Comment (Lee-W) [inline dags/ods/survey_cake/dags/questionnaire_2_bigquery.py:33]: nitpick: we could use `from __future__ import annotations` and replace it as the following

```suggestion
        FILENAMES: dict[str, dict] = {str(filepath): {}}
```

File: docs/CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+- Table name convention:
+  ![img](https://miro.medium.com/max/1400/1*bppuEKMnL9gFnvoRHUO8CQ.png)
+
+### Format
+
+Please use `make format` to format your code before committing, otherwise, the CI will fail.
+
+### Commit Message
+
+It is recommended to use [Commitizen](https://commitizen-tools.github.io/commitizen/).
```
Comment (Lee-W) [inline docs/CONTRIBUTING.md:75]: ❤️

File: docs/DEPLOYMENT.md [hunk 9]
```diff
@@ -0,0 +1,21 @@
+# Deployment Guide
+
+1. Login to the data team's server:
+    1. Run: `gcloud compute ssh --zone "asia-east1-b" "data-team" --project "pycontw-225217"`
+    2. Services:
+        * ETL: `/home/zhangtaiwei/pycon-etl`
+        * Metabase is located here: `/mnt/disks/data-team-additional-disk/pycontw-infra-scripts/data_team/metabase_server`
+
+2. Pull the latest codebase to this server: `sudo git pull`
```
Comment (Lee-W) [inline docs/DEPLOYMENT.md:9]: `sudo` looks dangerous. Just want to confirm whether it's required

File: requirements.txt (hunk 2 above)
Comment (henry410213028) [reply requirements.txt:12]: I try to remove it.

File: pyproject.toml (hunk 3 above)
Comment (henry410213028) [reply pyproject.toml:34]: Yes, this file should be removed to avoid any confusion.

Comment (henry410213028) [reply pyproject.toml]: @david30907d I need your help to fix the Python CI install dependencies step, as I don’t have permission to make changes.

File: README.md (hunk 6 above)
Comment (henry410213028) [reply README.md]: ok, I use venv module instead.

File: dags/ods/survey_cake/dags/questionnaire_2_bigquery.py (hunk 7 above)
Comment (henry410213028) [reply dags/ods/survey_cake/dags/questionnaire_2_bigquery.py]: thanks your suggestion.

File: docs/DEPLOYMENT.md (hunk 9 above)
Comment (henry410213028) [reply docs/DEPLOYMENT.md]: This is a issue, project path should be move to another path.

File: .env.template (hunk 5 above)
Comment (henry410213028) [reply .env.template:2]: Is AIRFLOW_HOME redundant?

File: pyproject.toml (hunk 3 above)
Comment (david30907d) [reply pyproject.toml]: > @david30907d I need your help to fix the Python CI install dependencies step, as I don’t have permission to make changes.

why is that, you're already admin of this repo 😂

**Comment by henry410213028** [comment]
upgrade airflow to v1.10.15, due to [v1.10.13](https://pypi.org/project/apache-airflow/1.10.13/) has been yanked.

File: .env.template (hunk 5 above)
Comment (Lee-W) [reply .env.template:2]: no, I mean BIGQUERY_PROJECT

File: docs/DEPLOYMENT.md (hunk 9 above)
Comment (Lee-W) [reply docs/DEPLOYMENT.md]: got it

File: .env.template (hunk 5 above)
Comment (henry410213028) [reply .env.template:2]: developer may use their own GCP project for testing

Comment (Lee-W) [reply .env.template:2]: Yep, that's why I think making it configurable might be a good idea. WDYT?

**Pull request merged**